command.go

```diff
@@ -80,9 +80,10 @@ func (c *Command) run(args []string) error {
 		// Parse flags
 		err := c.parseFlags(args)
 		if err != nil {
-			fmt.Printf("Error: %s\n\n", err.Error())
-			c.PrintHelp()
-			return err
+			if c.app.errorHandler != nil {
+				return fmt.Errorf("%s\n%s", err, c.app.errorHandler())
```

Comment:
should we keep it like that, meaning always output the flag error first and then users custom err or completely remove the original flag err? I'd think not cause we dont give the user a means to retrieve the undefined flag.

Reply:
ok sorry after working on it I realised, u meant to make the anonymous function `errorHandler` return an err?
```go
// Cli - The main application object.
type Cli struct {
	version        string
	rootCommand    *Command
	defaultCommand *Command
	preRunCommand  func(*Cli) error
	bannerFunction func(*Cli) string
	errorHandler   func() error
}
```

but what should be the best solution then, this?
```go
if c.app.errorHandler != nil {
	return fmt.Errorf("%s\n%s' for usage", err, c.app.errorHandler())
}
return fmt.Errorf("%s\nSee '%s --help' for usage", err, c.app.Name())
```

or
```go
if c.app.errorHandler != nil {
	return c.app.errorHandler()
}
return fmt.Errorf("%s\nSee '%s --help' for usage", err, c.app.Name())
```

I was thinking of something equivalent of `cli.OtherArgs()` but for flags, something like `cli.UndefinedFlag()` for use inside the errorHandler but that would create one more API endpoint + we need the removed function back.

I have no opinion here. Both the sensible minimal default  like. the full on custom I too like. I'll let u decide :)